tests/ui/rfcs/rfc-2632-const-trait-impl/feature-gate.rs

```diff
@@ -10,5 +10,13 @@ trait T {}
 impl const T for S {}
 //[stock]~^ ERROR const trait impls are experimental
 
+const fn f<A: ~const T>() {} //[stock]~ ERROR const trait impls are experimental
+fn g<A: const T>() {} //[stock]~ ERROR const trait impls are experimental
+
+macro_rules! discard { ($ty:ty) => {} }
+
+discard! { impl ~const T } //[stock]~ ERROR const trait impls are experimental
+discard! { impl const T } //[stock]~ ERROR const trait impls are experimental
```

Comment:
It looks like the following will regress with this PR but currently works in nightly:

```rust
macro_rules! check {
    ($ty:ty) => { compile_error!("....") };
    (impl $c:ident) => {  };
}

check! { impl const }

fn main() {}
```

But I don't think this would be a huge regression. Feel free to do a crater run if you are unsure.

Reply:
Very good catch! That looks pretty much unfixable 😢 I won't start a crater run since this doesn't look anything anybody would actually write and expect to pass compilation (tm).